app/models/concerns/civil_servant_searchable.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module CivilServantSearchable
+  WORD_SIMILARITY_QUERY = <<~SQL.squish
+    (
+      POWER(word_similarity(civil_servants.first_name, :term), 2) +
+      POWER(word_similarity(civil_servants.last_name, :term), 2) +
+      POWER(word_similarity(users.email, :term), 2) +
+      (CASE WHEN civil_servants.first_name ILIKE :begins_with_term THEN 0.5 ELSE 0 END) +
+      (CASE WHEN civil_servants.last_name ILIKE :begins_with_term THEN 0.5 ELSE 0 END)
+    )
+  SQL
+
+  WORD_SIMILARITY_SELECT_QUERY = <<~SQL.squish
+    #{WORD_SIMILARITY_QUERY} AS weight
+  SQL
+
+  WORD_SIMILARITY_WHERE_QUERY = <<~SQL.squish
+    #{WORD_SIMILARITY_QUERY} > 0.1
+  SQL
+
+  def search(term)
+    query_params = { term: term, begins_with_term: "#{term}%" }
+
+    select_query = sanitize_sql_for_assignment([WORD_SIMILARITY_SELECT_QUERY, query_params])
+    where_query = sanitize_sql_for_assignment([WORD_SIMILARITY_WHERE_QUERY, query_params])
+
+    CivilServant
+      .distinct
```

Comment:
Is the distinct needed? since email is unique anyway, this would have no effect. And it introduces a `O(n*log(n))` additional complexity...

Reply:
I was able to remove it here, had to add it in CivilServantSelect2Options (former: CivilServantSearch) on the user join. Otherwise I received duplicate entries in the result list